client/src/components/Homepage/index.js

```diff
@@ -37,7 +57,14 @@ const Homepage = () => {
   const [joinedSessionCode, setJoinedSessionCode] = useLocalStorage(
     'joinedSessionCode',
     ''
-  )
+    )
+    
+  /* state to toggle the navbar of the screen */
+  const [visible, setVisible] = useState(true)
+    
+  //State to change the themes
+  const [currentTheme, setCurrentTheme] = useState(CONSTANTS.DEFAULT_THEME) 
+    
   // non cached state vars
   const [output, setOutput] = useState('')
 
```

Comment:
```suggestion
    )
    const [currentTheme, setCurrentTheme] = useLocalStorage('currentTheme',CONSTANTS.DEFAULT_THEME)  //State to change the themes
    
  // non cached state vars
  const [output, setOutput] = useState('')
  const [visible, setVisible] = useState(true)  // state to toggle the navbar of the screen
```

Have a look at the suggested change. That is how things are structured. We have 2 kinds of state variables cached and non-cached. For caching, we have a custom hook call `useLocalstorage` defined which can be used directly.
You can find its definition here: https://github.com/nafees87n/codebox/blob/main/client/src/hooks/useLocalStorage.js

So manually handling the localstorage is not required as you have done here.